pkg/gitreceive/storage/object.go

```diff
@@ -72,3 +73,16 @@ func WaitForObject(statter ObjectStatter, bucketName, objKey string, tick, timeo
 		}
 	}
 }
+
+func DownloadObject(getter ObjectGetter, bucketName, objKey string) ([]byte, error) {
+	reader, err := getter.GetObject(bucketName, objKey)
+	if err == nil {
+		defer reader.Close()
+		if data, err := ioutil.ReadAll(reader); err != nil {
+			return nil, err
+		} else {
+			return data, err
+		}
+	}
```

Comment:
@kmala can you switch the boolean logic around, so that the `err != nil` logic is inside the `if`? It'll be more idiomatic that way.